### PR TITLE
chore: remove unused scene messages pool queue

### PR DESCRIPTION
**Pull request by pravusjif**
* Removed SceneController's sceneMessagesPool usage, since that queue is being infinitely filled with every "entity component" message generating the memory leak reported at https://github.com/decentraland/unity-renderer/issues/5532#issuecomment-1718661066
* Increased memory manager minimum memory check, as the current explorer supports 4gbs max instead of the old 2gb (and triggering the unload of assets always introduces a big hiccup)
* updated webgl max memory size to 4096 in project settings just in case (should already be overwritten by [emscripten linker flag](https://github.com/decentraland/unity-renderer/blob/4c5f3d00a02b3a2f9a04bf76f589066817da1813/unity-renderer/Assets/Scripts/Editor/PreBuildProcessing.cs#L24))
* Minor code improvement

--------------------------

### TESTING
This PR shoul be tested as thoroughly as a Release Candidate test.

--------------------------

### <samp>🤖 Generated by Copilot at a3ca31c</samp>

This pull request refactors the scene message handling logic in the `SceneController` class and its related interfaces. It removes the unused and inefficient `sceneMessagesPool` and simplifies the creation of `QueuedSceneMessage_Scene` instances. It also cleans up some unnecessary `using` directives and improves the code style.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs [hunk 1]
```diff
@@ -54,14 +54,11 @@ public void Initialize()
 
             DataStore.i.debugConfig.isDebugMode.OnChange += OnDebugModeSet;
 
-            SetupDeferredRunners();
+            // SetupDeferredRunners(); // TODO: is this needed ?? (WebGL/Desktop appears to run OK without it)
```
Comment (pravusjif) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs:57]: @Kinerius from what I tested (WebGL + Desktop) everything works fine without having this call, do you know if we really need this? if I can remove it I can also remove more code associated to that.

**Review by Kinerius — APPROVED** [review]
GFJ ❤️

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs (hunk 1 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs]: It was needed in the past when this was the cornerstone of the SDK6, I think the old MessageQueue system has to go

**Review by Kinerius — APPROVED** [review]

**Review by anicalbano — APPROVED** [review]
✅ Reviewed by QA

**Review by popuz — APPROVED** [review]
Amazing work 🦅

**Pull request merged**